### PR TITLE
Deprecate state-machine mixin

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/swarm/neo/util/StateMachine.d [hunk 1]
```diff
@@ -59,6 +59,8 @@ version ( UnitTest )
 
 *******************************************************************************/
 
+deprecated("Requests should be adapted to use one of the new request core "
+    "mixins in swarm.neo.client.mixin.")
```
Comment (mathias-lang-sociomantic) [inline src/swarm/neo/util/StateMachine.d]: Can you put this on the same line, even if it goes over 80 chars ?

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/StateMachine.d]: Sure. Why?

Comment (mathias-lang-sociomantic) [reply src/swarm/neo/util/StateMachine.d]: So it does not trigger a deprecation with the newest DMD (v2.071+)

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/StateMachine.d]: Gotcha.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/util/StateMachine.d]: Updated.

**Comment by nemanja-boric-sociomantic** [comment]
More deprecations for v5.x.x ;-)

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**